hera_cal/io.py

```diff
@@ -528,6 +528,11 @@ def read(self, bls=None, polarizations=None, times=None, time_range=None, lsts=N
                                  times=times, time_range=time_range, lsts=lsts, lst_range=lst_range, frequencies=frequencies, 
                                  freq_chans=freq_chans, read_data=read_data, run_check=run_check, check_extra=check_extra,
                                  run_check_acceptability=run_check_acceptability, **kwargs)
+                elif self.filetype == 'uvfits':
+                    super().read(self.filepaths, file_type='uvfits', axis=axis, bls=bls, polarizations=polarizations, times=times,
```

Comment:
This line can be merged with the equivalent the above read for uvh5. Then we can make `self.unphase_to_drift()` an addition interior `if`.